src/Microsoft.AspNetCore.Razor.Language/RazorTemplateEngine.cs

```diff
@@ -100,6 +95,38 @@ public RazorCSharpDocument GenerateCode(RazorProjectItem projectItem)
             return GenerateCode(codeDocument);
         }
 
+        public virtual RazorCSharpDocument GenerateCode(
+            Func<Stream> streamFactory, 
+            string relativePhysicalPath,
```

Comment:
Consider `<Content Include="Dir1\Dir2\Index.cshtml" Link="Dir3\NotIndex.cshtml" />`. The intent is to treat the file as if it is was located at `Dir3\NotIndex.cshtml` but read it from it's source location `Dir1\Dir2\Index.cshtml`. i.e. we need to be able to specify `filePath` from the task. We can figure out where this method needs to live, but I was just setting up my expectation

Reply:
I don't think I'd change this class to add support for linked files. This should be part of the RazorProject default implementation. That implementation should be able to have an internal understanding of `FilePath => ProjectItem` and then the ProjectItem itself can have an understanding of if it should read content from a different location.

Reply:
Agree with @NTaylorMullen. I anticipated this, and it needs to be part of the filesystem.

Reply:
We will also need to flow the same information in through the Razor project system into the IDE.